### PR TITLE
Fixes #161 - close jdbc statements and result sets

### LINKED ISSUE #161: NetCDF Generator consumes excessive memory
Looks like the NetCDF generator is not closing some result sets resulting in excessive memory consumption including out of memory errors in production.

Refer https://github.com/aodn/internal-discussions/issues/274

### PR DESCRIPTION
**Pull request by jonescc**

### CODE REVIEW

File: src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java [hunk 1]
```diff
@@ -290,60 +291,70 @@ private void populateValues(
...
 
-    public void closeQuietly() {
+    public void close() {
+        closeQuietly(featureInstancesRS);
+        closeQuietly(featuresStmt);
+        closeQuietly(conn);
+    }
+
+    private void closeQuietly(AutoCloseable autoCloseable) {
+        if (autoCloseable == null) return;
```
Comment (julian1) [inline src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java]: Under what condition would autoCloseable be null?

Comment (jonescc) [inline src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java]: if there's an exception thrown before actually creating the result set for example

File: src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java [hunk 2]
```diff
@@ -290,60 +291,70 @@ private void populateValues(
...
+
         try {
-            conn.close();
-        }
-        catch (SQLException e) {
-            logger.info("problem closing transaction");
+            autoCloseable.close();
+        } catch (Exception e) {
+            logger.warn(String.format("Problem closing %s", autoCloseable.getClass().getCanonicalName()), e);
```
Comment (julian1) [inline src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java]: What is the result of evaluating, 

```
String.format("Problem closing %s", autoCloseable.getClass().getCanonicalName())
```

Isn't it just 'AutoCloseable'

Is the actual exception given by 'e' reported in the logger.warn statement?

Comment (julian1) [inline src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java]: Ahh nevermind. Misunderstood.

**Pull request merged**